pennylane/_qubit_device.py

```diff
@@ -483,8 +477,18 @@ def sample_basis_states(self, number_of_states, state_probability):
         Returns:
             List[int]: the sampled basis states
         """
+        if self.shots is None:
+            warnings.warn(
+                "The number of shots have to be explicitly set on the device "
+                "when using sample-based measurements. Since no shots were specified, "
+                "a default of 1000 samples was used.",
```

Comment:
(minor) Just a preference to writing in present tense.
```suggestion
                "when using sample-based measurements. Since no shots are specified, "
                "a default of 1000 shots is used.",
```